src/Transform/ONNX/Rewrite.cpp

```diff
@@ -82,6 +82,25 @@ DenseElementsAttr insertZerosForNonPaddedDims(
   return rewriter.getI64TensorAttr(llvm::makeArrayRef(pads));
 }
 
+
+// Create an DenseElementsAttr of ArrayAttr.
+// This function is used to get Value Type for Scaler function.
+DenseElementsAttr createDenseArrayAttr(
+    PatternRewriter &rewriter, ArrayAttr origAttrs) {
+  mlir::Type elementType = rewriter.getF32Type();
+  int nElements = origAttrs.getValue().size();
+  SmallVector<float, 4> wrapper(nElements, 0);
+  if (origAttrs) {
+    for (int i = 0; i < nElements; ++i) {
+      wrapper[i] = origAttrs.getValue()[i].cast<FloatAttr>().getValueAsDouble();
```

Comment:
Why asDouble? Should be depended on element type.

Reply:
Similar to above, since this function is used to transfer scaler function's attributes (which only accept float value), I directly assume the values' types are in float form. I will have to check for the value type to cast and get value properly otherwise. Should I make that change here?